### PR TITLE
iop colorspace: code refactoring - clean-up

### LINKED ISSUE #14569: iop colorspace: code refactoring - clean-up
Would be good to replace `int` by `dt_iop_colorspace_type_t` for `[default/input/ouput]_colorspace` in all IOPs and the `iop_api.h`. There is also probably some corresponding changes to do in structs.

### PR DESCRIPTION
**Pull request by MStraeten**
replaced int by dt_iop_colorspace_type_t for [default/input/ouput/blend]_colorspace in all IOPs and the iop_api.h
moved dt_iop_colorspace_type_t definition into common/colorspaces.h

fixes #14569

### CODE REVIEW

File: src/iop/ashift.c [hunk 1]
```diff
@@ -157,7 +157,7 @@ int operation_tags_filter()
   return IOP_TAG_DECORATION | IOP_TAG_CLIPPING;
 }
 
-int default_colorspace(dt_iop_module_t *self,
+dt_iop_colorspace_type_t default_colorspace(dt_iop_module_t *self,
```
Comment (TurboGit) [inline src/iop/ashift.c:160]: The parameters should be aligned:

```
dt_iop_colorspace_type_t default_colorspace(dt_iop_module_t *self,
                                            dt_dev_pixelpipe_t *pipe,
                                            dt_dev_pixelpipe_iop_t *piece)
```

Many other instances below.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Thanks! This will be the first merge for 4.6.

File: src/iop/ashift.c (hunk 1 above)
Comment (MStraeten) [reply src/iop/ashift.c:160]: sed doesn't care about formatting ;)
will be done in some further steps

Comment (MStraeten) [reply src/iop/ashift.c:160]: aligning parameters done

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**